### DISCUSSION TITLE
iPXE boot fails to start

### DISCUSSION THREAD

**Discussion by cskwrd** [flatcar/Flatcar#1491] [Q&A] [answered]
I am trying to install Flatcar on a new bare metal x86_64 device. The boot process fails near instantly with `systemd[1]: Failed to isolate default target: Unit sysusr-usr.mount not found.` I thought secure boot might have been enable but the setup screen indicates that it is disabled. Is UEFI not supported on bare metal?

**Comment by tormath1** [comment]
Hello @cskwrd can you provide some information on how you installed things? How many memory do you have? Do you see other lines before the `systemd[1]: Failed to isolate default target: Unit sysusr-usr.mount not found.`? 

Thanks!

**Reply by cskwrd** [reply]
@tormath1 the device I'm working with is a beelink mini PC with 8GB of RAM and a 512GB nvme ssd.

I've tried following the docs and booting from iPXE "manually." When attempting to not manually, I've used the current image as well as the previous version's image. I've also tried using [netboot.xyz](https://netboot.xyz). I've _not_ tried using the the ISO.

The boot process makes it a bit further when I do not provide an ignition config, but I'm still left with an emergency shell.

I have also tried to boot other operating systems. I've been able to boot Arch Linux, and openSUSE tumbleweed.

Below is a screenshot of the boot screen:
![boot_screen_output.jpg](https://github.com/flatcar/Flatcar/assets/10216235/c179246e-3c9c-40f9-a1bc-68272a28c730)

Note: In the picture above I used netboot.xyz, and provided a `ignition.config.url`.

**Reply by jepio** [reply]
do you have any way of collecting the whole boot output (maybe a separate serial port?)

whats your ipxe boot script? The important thing is this: `initrd=flatcar_production_pxe_image.cpio.gz` kernel argument, without it the boot definitely fails.

**Reply by jepio** [reply]
there is also some issue in netboot that prevents some versions from working: https://github.com/flatcar/Flatcar/issues/994

**Comment by cskwrd** [comment]
I have tried a few different iterations of a bootscript.
```
#!ipxe

set base-url http://stable.release.flatcar-linux.net/amd64-usr/current

kernel ${base-url}/flatcar_production_pxe.vmlinuz initrd=flatcar_production_pxe_image.cpio.gz flatcar.first_boot=1 ignition.config.url=http://192.168.1.74:8080/start-ssh.ign
initrd ${base-url}/flatcar_production_pxe_image.cpio.gz
boot
```
The above is taken nearly verbatim from the flatcar docs.

```
#!ipxe

set base-url http://stable.release.flatcar-linux.net/amd64-usr/3815.2.3

kernel ${base-url}/flatcar_production_pxe.vmlinuz initrd=flatcar_production_pxe_image.cpio.gz flatcar.first_boot=1 ignition.config.url=http://192.168.1.74:8080/start-ssh.ign
initrd ${base-url}/flatcar_production_pxe_image.cpio.gz
boot
```
This one is trying an older version of flatcar.

[netboot.xyz bootscript template](https://github.com/netbootxyz/netboot.xyz/blob/2.0.79/roles/netbootxyz/templates/menu/flatcar.ipxe.j2)

When I try to boot using netboot.xyz and skip the `ignition.config.url` I get behavior similar to #994. The messages that the boot flips between are slightly different.

At this time I have no other way of grabbing the boot messages. This isn't a dev board so I don't have any dev header pins or test points. Is there a way to send the messages over another USB port using kernel parameters?

**Reply by ader1990** [reply]
Hello @cskwrd -> the issue you might be seeing can be related to the ignition config not being retrieved. I assume you have a DHCP server with an automatic or static configuration for the Beelink net card. Can you confirm that the DHCP IP of the Beelink net card is in the same network with `192.168.1.74` or `192.168.1.74` is accessible from that subnet and also share the content of the http://192.168.1.74:8080/start-ssh.ign?

**Reply by cskwrd** [reply]
Hi @ader1990 

I thought that might have been a problem so I hosted my bootscripts on the same web server. I can obviously download the bootscripts. According to the apache logs, no request is made for the ignition config.

Here is my ignition config (I have replaced my actual key):
```json
{"ignition":{"version":"3.3.0"},"passwd":{"users":[{"name":"core","sshAuthorizedKeys":["ssh-ed25519 keystuff"]}]}}
```

**Comment by cskwrd** [comment, marked as answer]
I was poking around with this some more. I think it is in fact an issue with netboot.xyz. If I perform the same steps using a kvm virtual machine, and try loading the same config files there, the boot also fails. I have tried using the the [pre-built iPXE ISO](http://boot.ipxe.org/ipxe.iso) directly, and it boots flatcar just fine.

It's not as simple as netboot.xyz, but that isn't relevant here. Thanks everyone for their time.
